.circleci/cimodel/pytorch_build_definitions.py

```diff
@@ -144,70 +143,63 @@ def gen_build_name(self, _):
         return self.name
 
 
-xenial_parent_config = Conf(
-    "xenial",
-    ["py3"],
-    pyver="3.6",
-    cuda_version="8",
-    gpu_resource="medium",
-)
-
-
-# TODO This is a short-term hack until it is converted to recursive tree traversal
-xenial_dependent_configs = [
-    Conf("xenial",
-         ["py3", DockerHide("multigpu")],
-         pyver="3.6",
-         cuda_version="8",
-         restrict_phases=["test"],
-         gpu_resource="large",
-         parent_build=xenial_parent_config,
-         ),
-    Conf("xenial",
-         ["py3", DockerHide("NO_AVX2")],
-         pyver="3.6",
-         cuda_version="8",
-         restrict_phases=["test"],
-         gpu_resource="medium",
-         parent_build=xenial_parent_config,
-         ),
-    Conf("xenial",
-         ["py3", DockerHide("NO_AVX"), DockerHide("NO_AVX2")],
-         pyver="3.6",
-         cuda_version="8",
-         restrict_phases=["test"],
-         gpu_resource="medium",
-         parent_build=xenial_parent_config,
-         ),
-
-    HiddenConf("pytorch_short_perf_test_gpu", parent_build=xenial_parent_config),
-    HiddenConf("pytorch_doc_push", parent_build=xenial_parent_config),
-]
+# TODO Convert these to graph nodes
+def gen_dependent_configs(xenial_parent_config):
+
+    extra_parms = [
+        (["multigpu"], "large"),
+        (["NO_AVX2"], "medium"),
+        (["NO_AVX", "NO_AVX2"], "medium"),
+    ]
 
+    configs = []
+    for parms, gpu in extra_parms:
 
-xenial_parent_config.dependent_tests = xenial_dependent_configs
+        c = Conf(
+            "xenial",
+            ["py3"] + parms,
+            pyver="3.6",
+            cuda_version="8",
+            restrict_phases=["test"],
+            gpu_resource=gpu,
+            parent_build=xenial_parent_config,
+        )
 
+        configs.append(c)
 
-# TODO this hierarchy is a work in progress
+    for x in ["pytorch_short_perf_test_gpu", "pytorch_doc_push"]:
+        configs.append(HiddenConf(x, parent_build=xenial_parent_config))
```

Comment:
Is a HiddenConf a "conf that is hidden behind its job name and should be renamed to say exactly what it's doing"?